### DISCUSSION TITLE
Raw input for Linux

### DISCUSSION THREAD

**Discussion by leath-dub** [H-M-H/Weylus#307] [Ideas]
I have a project that allows you to proxy raw android device events through linux uinput device: https://github.com/leath-dub/droidux

Wondering if this kind of thing would fit into weylus as it really improves latency.

I reads a device description from `adb getevent -pi` and creates a device directly. Then it proxies the events from adb directly to the uinput device.

**Comment by H-M-H** [comment]
Hmm, do you experience latency issues with Weylus? Weylus sends input events as JSON over a websocket connection and then feeds them to uinput (at least on Linux).

At least for me input latency has never been an issue, only the latency of the video stream is noticable to me.

**Reply by leath-dub** [reply]
It is very noticeable for me at least. I don't know how other people feel though.

Edit: enough to where I stopped using it and basically gave up trying to use my tablet on linux. That was until I sat down and said I would fix it, in the process I just found that you can use adb and getevent to directly proxy input devices.

**Reply by H-M-H** [reply]
Interesting, have you tried using `adb reverse` as described in the Readme, perhaps the issue is only network latency?

**Reply by leath-dub** [reply]
> Interesting, have you tried using `adb reverse` as described in the Readme, perhaps the issue is only network latency?

Better, but still not bearable for me. Also the stylus hover does not work. I understand the hover is a matter of browser support, but even so having a fallback that is lower latency and basically fool proof as it just publishes the raw events directly as they are read on the device directly is good I think.

Realistically I know a lot of people use weylus, and I think they would benefit from this, my project is too obscure and specific to stand on its own.
